src/ess/dream/io.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+import os
+from typing import Any, Dict, Union
+
+import scipp as sc
+import scippnexus as snx
+
+
+def load_nexus(
+    path: Union[str, os.PathLike],
+    *,
+    load_pixel_shape: bool = False,
+    entry: str = 'entry',
+    fold_detectors: bool = True,
+) -> sc.DataGroup:
+    """
+    Load an unprocessed DREAM NeXus file.
+
+    See https://confluence.esss.lu.se/pages/viewpage.action?pageId=462000005
+    and the ICD DREAM interface specification for details.
+
+    Notes (2023-12-06):
+
+    - Mounting-unit, cassette, and counter roughly correspond to the azimuthal angle
+      in the mantle detector. However, counter is reversed in the current files. This
+      may also be the case in the other detectors.
+    - The endcap detectors have a irregular structure that cannot be fully folded.
+      This is not a problem but note again that the counter may be reversed. It is
+      currently not clear if this is a bug in the file.
+    - The high-resolution detector has a very odd numbering scheme. The SANS detector
+      is using the same, but is not populated in the current files. The scheme
+      attempts to follows some sort of physical ordering in space (x,y,z), but it
+      looks partially messed up.
+
+    Parameters
+    ----------
+    path:
+        Path to the NeXus file.
+    load_pixel_shape:
+        If True, load the pixel shape from the file's NXoff_geometry group. This is
+        often unused by would slow down file loading. Default is False.
+    entry:
+        Name of the entry to load. Default is "entry".
+    fold_detectors:
+        If True, fold the detector data to (partially) mimic the logical detector
+        structure. Default is True.
+
+    Returns
+    -------
+    :
+        A data group with the loaded file contents.
+    """
+    definitions = snx.base_definitions()
+    if not load_pixel_shape:
+        definitions["NXdetector"] = FilteredDetector
+
+    with snx.File(path, definitions=definitions) as f:
+        dg = f[entry][()]
+    dg = snx.compute_positions(dg)
+    if not fold_detectors:
+        return dg
+    instrument = dg['instrument']
+    mantle = instrument['mantle_detector']
+    mantle['mantle_event_data'] = mantle['mantle_event_data'].fold(
+        dim='detector_number',
+        sizes={
+            'wire': 32,
+            'mounting_unit': 5,
+            'cassette': 6,
+            'counter': 2,
+            'strip': 256,
```

Comment:
These names correspond to the ICD, not GEANT4 names on the confluence page. Are these the ones we want to go with? Should we rename the dims in the GEANT4 loader?

Reply:
ok